File: taxi/dmp/spark/spark-over-yt/yt-wrapper/src/main/scala/ru/yandex/spark/yt/wrapper/operation/OperationStatus.scala
package ru.yandex.spark.yt.wrapper.operation

sealed abstract class OperationStatus(val isFinished: Boolean,
                                      val isSuccess: Boolean,
                                      val name: String)

object OperationStatus {
  case object Unknown extends OperationStatus(false, false, "unknown")

  case object Starting extends OperationStatus(false, false, "starting")

  case object Initializing extends OperationStatus(false, false, "initializing")

  case object Preparing extends OperationStatus(false, false, "preparing")

  case object Materializing extends OperationStatus(false, false, "materializing")

  case object Reviving extends OperationStatus(false, false, "reviving")

  case object ReviveInitializing extends OperationStatus(false, false, "revive_initializing")

  case object Orphaned extends OperationStatus(false, false, "orphaned")

  case object WaitingForAgent extends OperationStatus(false, false, "waiting_for_agent")

  case object Pending extends OperationStatus(false, false, "pending")

  case object Running extends OperationStatus(false, false, "running")

  case object Completing extends OperationStatus(false, false, "completing")

  case object Completed extends OperationStatus(true, true, "completed")

  case object Aborting extends OperationStatus(false, false, "aborting")

  case object Aborted extends OperationStatus(true, false, "aborted")

  case object Failing extends OperationStatus(false, false, "failing")

  case object Failed extends OperationStatus(true, false, "failed")

  def getByName(name: String): OperationStatus = {
    Seq(Unknown, Starting, Initializing, Preparing,
      Materializing, Reviving, ReviveInitializing,
      Orphaned, WaitingForAgent, Pending, Running,
      Completing, Completed, Aborting, Aborted, Failing)
      .find(_.name == name.toLowerCase)
      .getOrElse(throw new IllegalArgumentException(s"Unknown operation status $name"))
  }
}

File: ytsaurus/yt/java/ytsaurus-client/src/main/java/tech/ytsaurus/client/operations/OperationStatus.java
package tech.ytsaurus.client.operations;

import tech.ytsaurus.core.StringValueEnum;
import tech.ytsaurus.core.StringValueEnumResolver;
import tech.ytsaurus.lang.NonNullApi;
import tech.ytsaurus.lang.NonNullFields;

@NonNullApi
@NonNullFields
public enum OperationStatus implements StringValueEnum {

    UNKNOWN(false, false, "unknown"),
    STARTING(false, false, "starting"),
    INITIALIZING(false, false, "initializing"),
    PREPARING(false, false, "preparing"),
    MATERIALIZING(false, false, "materializing"),
    REVIVING(false, false, "reviving"),
    REVIVE_INITIALIZING(false, false, "revive_initializing"),
    ORPHANED(false, false, "orphaned"),
    WAITING_FOR_AGENT(false, false, "waiting_for_agent"),
    PENDING(false, false, "pending"),
    RUNNING(false, false, "running"),

    SUSPENDED(false, false, "suspended"),

    COMPLETING(false, false, "completing"),
    COMPLETED(true, true, "completed"),

    ABORTING(false, false, "aborting"),
    ABORTED(true, false, "aborted"),

    FAILING(false, false, "failing"),
    FAILED(true, false, "failed");

    public static final StringValueEnumResolver<OperationStatus> R = StringValueEnumResolver.of(OperationStatus.class);

    private final boolean finished;
    private final boolean success;

    private final String value;

    OperationStatus(boolean finished, boolean success, String value) {
        this.finished = finished;
        this.success = success;
        this.value = value;
    }

    public boolean isFinished() {
        return finished;
    }

    public boolean isSuccess() {
        return success;
    }

    @Override
    public String value() {
        return value;
    }

}
